server/src/main/kotlin/org/javacs/kt/rename/Rename.kt

```diff
@@ -7,7 +7,7 @@ import org.javacs.kt.SourcePath
 import org.javacs.kt.references.findReferences
 
 fun renameSymbol(file: CompiledFile, cursor: Int, sp: SourcePath, newName: String): WorkspaceEdit? {
-    val (declaration, location) = file.findDeclaration(cursor) ?: return null
+    val (declaration, location) = file.findDeclaration(cursor) ?: file.findDeclarationCursorSite(cursor) ?: return null
```

Comment:
Since this fallback seems to be more generally useful, perhaps it would be worth moving it directly into `findDeclaration`? Or perhaps rename `CompiledFile.findDeclaration` -> `findDeclarationReference` and then define something along the lines of

```kotlin
fun findDeclaration(cursor: Int) = findDeclarationReference(cursor) ?: findDeclarationCursorSite(cursor)
```

as a method in `CompiledFile`.

Reply:
Good idea! 🙂 Done in a57d8a8